apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2137,31 +2140,60 @@ def create_ingestion_dag(
             input_vectors_work_items_per_worker_during_sampling
         )
 
-        # We can't set as default in the function due to the use of `str(threads)`
-        # For consistency we then apply all defaults for resources here.
-        if ingest_resources is None:
-            ingest_resources = {"cpu": str(threads), "memory": "16Gi"}
 
-        if consolidate_partition_resources is None:
-            consolidate_partition_resources = {"cpu": str(threads), "memory": "16Gi"}
+        if(size >= input_vectors_per_work_item):
+            # We can't set as default in the function due to the use of `str(threads)`
+            # For consistency we then apply all defaults for resources here.
+            if ingest_resources is None:
+                ingest_resources = {"cpu": str(threads), "memory": "16Gi"}
 
-        if copy_centroids_resources is None:
-            copy_centroids_resources = {"cpu": "1", "memory": "2Gi"}
+            if consolidate_partition_resources is None:
+                consolidate_partition_resources = {"cpu": str(threads), "memory": "16Gi"}
 
-        if random_sample_resources is None:
-            random_sample_resources = {"cpu": "2", "memory": "6Gi"}
+            if copy_centroids_resources is None:
+                copy_centroids_resources = {"cpu": "1", "memory": "2Gi"}
 
-        if kmeans_resources is None:
-            kmeans_resources = {"cpu": "8", "memory": "32Gi"}
+            if random_sample_resources is None:
+                random_sample_resources = {"cpu": "2", "memory": "6Gi"}
 
-        if compute_new_centroids_resources is None:
-            compute_new_centroids_resources = {"cpu": "1", "memory": "8Gi"}
+            if kmeans_resources is None:
+                kmeans_resources = {"cpu": "8", "memory": "32Gi"}
 
-        if assign_points_and_partial_new_centroids_resources is None:
-            assign_points_and_partial_new_centroids_resources = {
-                "cpu": str(threads),
-                "memory": "12Gi",
-            }
+            if compute_new_centroids_resources is None:
+                compute_new_centroids_resources = {"cpu": "1", "memory": "8Gi"}
+
+            if assign_points_and_partial_new_centroids_resources is None:
+                assign_points_and_partial_new_centroids_resources = {
+                    "cpu": str(threads),
+                    "memory": "12Gi",
+                }
+        else:
+            scale_factor = input_vectors_per_work_item/size
+            print("Scale factor: ", scale_factor)
```

Comment:
Avoid printing to stdout, if you want this for debug purposes use `logger.debug` instead.